tests/Valit.Tests/NestedObject/Nested_Object_Collection_Validation_Tests.cs

```diff
@@ -0,0 +1,145 @@
+            });
+            
+            exception.ShouldBeOfType(typeof(ValitException));
+        }
+
+        [Fact]
+        public void Validation_With_Complete_Strategy_Fails_When_Invalid_Collection_Is_Given()
+        {
+            var rootObject = RootObject.GetInvalid();
+
+            var result = ValitRules<RootObject>
+                .Create()
+                .Ensure(m => m.StringValue, _=>_
+                    .Required())
+                .EnsureFor(m => m.NestedObjectCollection, new NestedObjectRulesProvider())
+                .For(rootObject)
+                .Validate();
+
+            Assert.False(result.Succeeded);
+            result.ErrorMessages.Where(m => m == "One").Count().ShouldBe(1);
+            result.ErrorMessages.Where(m => m == "Two").Count().ShouldBe(1);
+            result.ErrorMessages.Where(m => m == "Three").Count().ShouldBe(2);
+        }
+
+        [Fact]
+        public void Validation_With_FailFast_Strategy_Fails_When_Invalid_Collection_Is_Given()
+        {
+            var rootObject = RootObject.GetInvalid();
+            
+            var result = ValitRules<RootObject>
+                .Create()
+                .WithStrategy(picker => picker.FailFast)
+                .Ensure(m => m.StringValue, _=>_
+                    .Required())
+                .EnsureFor(m => m.NestedObjectCollection, new NestedObjectRulesProvider())
+                .For(rootObject)
+                .Validate();
+
+            Assert.False(result.Succeeded);
+            result.ErrorMessages.Count().ShouldBe(1);
+            result.ErrorMessages.First().ShouldBe("Two");
+        }
+
+        [Fact]
+        public void Validation_With_Complete_Strategy_Succeeds_When_Valid_Collection_Is_Given()
+        {
+            var rootObject = RootObject.GetValid();
+            
+            var result = ValitRules<RootObject>
+                .Create()
+                .Ensure(m => m.StringValue, _=>_
+                    .Required())
+                .EnsureFor(m => m.NestedObjectCollection, new NestedObjectRulesProvider())
+                .For(rootObject)
+                .Validate();
+
+            Assert.True(result.Succeeded);
+        }
+
+        [Fact]
+        public void Validation_With_FailFast_Strategy_Succeeds_When_Valid_Collection_Is_Given()
+        {
+            var rootObject = RootObject.GetValid();
+            
+            var result = ValitRules<RootObject>
+                .Create()
+                .WithStrategy(picker => picker.FailFast)                
+                .Ensure(m => m.StringValue, _=>_
+                    .Required())
+                .EnsureFor(m => m.NestedObjectCollection, new NestedObjectRulesProvider())
+                .For(rootObject)
+                .Validate();
+
+            Assert.True(result.Succeeded);
+        }
+
+#region  ARRANGE
+        class RootObject
+        {
+            public string StringValue { get; set; }
```

Comment:
Do we need the `StringValue` for these tests, since we're testing `NestedObjectCollection` here?

Reply:
removed